extensions/wikia/FacebookClient/scripts/FacebookClient.XFBML.js

```diff
@@ -2,19 +2,34 @@
 
 /**
  * Render FB XML tags emebedded into wikitext
- * This file is loaded via ResourceLoader only when FB XML tags are present on the page
  */
 $(function () {
 	'use strict';
 
 	// load api on page load and on demand
 	mw.hook('wikipage.content').add(function ($content) {
-		$.loadFacebookAPI()
-			.done(function () {
-				$('.sso-login').removeClass('hidden');
 
-				// scan the new content for any fb tags
-				FB.XFBML.parse($content[0]);
-			});
+		if ((xfbmlTagsOnPage())) {
+			if (facebookSDKNotLoaded()) {
+				// XFBML tags rendered automatically when SDK loads
+				$.loadFacebookSDK();
+			} else {
+				renderXFBMLTags();
```

Comment:
The original implementation relied on done() to fire `FB.XFBML.parse($content[0]);`. There were a couple of problems with this:

1.) The config object we pass to FB.init [here](https://github.com/Wikia/app/blob/dev/resources/wikia/libraries/jquery/loadLibrary/jquery.wikia.loadExternalLibrary.js#L63) is already instructing FB to parse the XFBML tags. Calling it manually as well was causing some race condition where the first XFBML tag would never render.
2.) The way that that `$.loadFacebookSDK()` works if the SDK is already loaded then the resolve on the promise is never fired. I tried to a couple different methods for fixing that, but none worked particularly well. I figured it's a pretty edge case need, so I could just keep the logic out here.